library/src/main/java/bio/terra/pfb/PfbReader.java

```diff
@@ -48,23 +48,32 @@ public static String showMetadata(String fileLocation) throws IOException {
   }
 
   public static List<String> show(String fileLocation) throws IOException {
-    GenericDatumReader<GenericRecord> datumReader = new GenericDatumReader<>();
-    URL url = isValidUrl(fileLocation);
-    GenericRecord genericRecord = null;
     List<String> data = new ArrayList<>();
-    try (InputStream in =
-            url != null ? readFromSignedUrl(url.toString()) : readFromLocalFile(fileLocation);
-        DataFileStream<GenericRecord> reader = new DataFileStream<>(in, datumReader)) {
-      // Skip Metadata Object, which should always appear first
-      reader.next(genericRecord);
-      while (reader.hasNext()) {
-        genericRecord = reader.next(genericRecord);
-        data.add(convertEnum(genericRecord.toString()));
+    try {
+      DataFileStream<GenericRecord> records = PfbReader.getGenericRecordsStream(fileLocation);
+
+      while (records.hasNext()) {
+        data.add(convertEnum(records.next().toString()));
       }
       return data;
+    } catch (IOException e) {
+      throw new InvalidPfbException("Error reading PFB Value object");
     }
   }
 
+  public static DataFileStream<GenericRecord> getGenericRecordsStream(String fileLocation)
+      throws IOException {
+    GenericDatumReader<GenericRecord> datumReader = new GenericDatumReader<>();
+    URL url = isValidUrl(fileLocation);
+
+    InputStream in =
+        url != null ? readFromSignedUrl(url.toString()) : readFromLocalFile(fileLocation);
+    var reader = new DataFileStream<>(in, datumReader);
+    // advance past metadata
+    reader.next();
+    return reader;
```

Comment:
I think the `InputStream` should be inside a try-with-resources to make sure it gets closed:

```suggestion
    try (InputStream in =
                 url != null ? readFromSignedUrl(url.toString()) : readFromLocalFile(fileLocation)) {
      var reader = new DataFileStream<>(in, datumReader);
      // advance past metadata
      reader.next();
      return reader;
    }
```

Reply:
on second thought, this will probably close the inputstream before we mean to - it could cause functional problems :(

Reply:
haha yup! I also thought that but couldnt get it to work that way (the stream would work sort of but would close unexpectedly before it would get to the end) 

adding a simple try catch around this wouldn't hurt though, ill do that. and tests :D

Reply:
Even though we can't use `try with resources` here, I think it'd be reasonable to expect client code to do so with the returned `DataFileStream` -- maybe it'd be worth a unit test that explicitly uses `try with resources` to verify the `DataFileStream` is able to be closed correctly using this mechanism?

Reply:
ah missed this -- honestly it seems like Java does the right thing for us regardless but happy to add another test if you think it is necessary.